zanata-war/src/main/java/org/zanata/adapter/XliffAdapter.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors as indicated by the
+ * @author tags. See the copyright.txt file in the distribution for a full
+ * listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+
+package org.zanata.adapter;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.zanata.adapter.xliff.XliffCommon;
+import org.zanata.adapter.xliff.XliffReader;
+import org.zanata.adapter.xliff.XliffWriter;
+import org.zanata.common.LocaleId;
+import org.zanata.exception.FileFormatAdapterException;
+import org.zanata.rest.dto.resource.Resource;
+import org.zanata.rest.dto.resource.TranslationsResource;
+
+import com.google.common.base.Optional;
+import org.zanata.util.FileUtil;
+
+/**
+ * Adapter for read and write {@link org.zanata.common.DocumentType#XLIFF} file
```

Comment:
- [ ] .(grammar) "... for reading and writing ..." or "... to read and write ..."